### PR TITLE
Add -o json flag for linkerd multicluster gateways

### LINKED ISSUE #8285: Introduce `-o json` flag for `linkerd multicluster gateways`
### What problem are you trying to solve?

Having JSON output options for commands helps with programatic use of the CLI. The main driver behind this feature request though is for use in tests so that we can eliminate the manual parsing that we currently have to do in the integration tests as noted [here](https://github.com/linkerd/linkerd2/pull/8139#discussion_r840698514).

### How should the problem be solved?

We can look at other `-o json` flag implementations (for example `linkerd multicluster check`) as it should be similar.

### Any alternatives you've considered?

No

### How would users interact with this feature?

`linkerd multicluster gateways -o json`

### Would you like to work on this feature?

_No response_

### PR DESCRIPTION
**Pull request by hiteshwani29**
Added flag -o json to show multicluster gateways output in the json format

Fixes #8285

Signed-off-by: Hitesh Wani [hiteshwani29@gmail.com](mailto:hiteshwani29@gmail.com)

Attaching screenshot of local testing
Screenshot from 2023-04-18 17-48-56

### CODE REVIEW

**Review by adleong — COMMENTED** [review]
One tiny suggestion, but otherwise this looks great!  Thanks!

**Comment by adleong** [comment]
You'll also need to merge the latest main to pick up fixes to the failing tests.

File: multicluster/cmd/gateways.go [hunk 1]
```diff
@@ -127,19 +130,34 @@ func newGatewaysCommand() *cobra.Command {
...
+			switch opts.output {
+			case "":
+				renderGateways(statuses, stdout)
+			case "json":
+				out, err := json.MarshalIndent(statuses, "", "  ")
+				if err != nil {
+					fmt.Fprint(os.Stderr, err)
+					os.Exit(1)
+				}
+				fmt.Printf("%s", out)
```
[the remark this answers is not in the document]
Comment (hiteshwani29) [reply multicluster/cmd/gateways.go]: Done @adleong

**Comment by hiteshwani29** [comment]
@adleong can you relook at the PR again?

File: multicluster/cmd/gateways.go [hunk 2]
```diff
@@ -127,19 +130,34 @@ func newGatewaysCommand() *cobra.Command {
...
+					gatewayStatus.Latency = uint64(metrics.GetGauge().GetValue())
 					break
 				}
 
 				statuses = append(statuses, gatewayStatus)
 			}
-			renderGateways(statuses, stdout)
+
+			switch opts.output {
+			case "":
```
Comment (mateiidavid) [inline multicluster/cmd/gateways.go:141]: I think we should maybe treat this in a similar way as we do in the viz counterpart where the output can either be "table", or "json", with the default being "table". https://github.com/linkerd/linkerd2/blob/main/viz/cmd/stat.go

File: multicluster/cmd/gateways.go [hunk 3]
```diff
@@ -31,10 +34,10 @@ type (
 	}
 
 	gatewayStatus struct {
-		clusterName      string
-		alive            bool
-		numberOfServices int
-		latency          uint64
+		ClusterName      string `json:"clusterName"`
```
Comment (mateiidavid) [inline multicluster/cmd/gateways.go:37]: I guess the visibility changed here because we want to marshal it? 🤔

Comment (hiteshwani29) [reply multicluster/cmd/gateways.go:37]: Yes right, @mateiidavid 
Without this, we are not able to marshal it to JSON fields.

File: multicluster/cmd/gateways.go (hunk 2 above)
Comment (hiteshwani29) [reply multicluster/cmd/gateways.go]: Done @mateiidavid 
Added table output in the default.

**Comment by hiteshwani29** [comment]
@mateiidavid Thanks for reviewing the PR :pray: 
I made the suggested changes. Can you please review the PR again?

**Review by adleong — APPROVED** [review]

**Review by mateiidavid — APPROVED** [review]

**Pull request merged**